protocol/messenger_handler.go

```diff
@@ -1443,10 +1443,38 @@ func (m *Messenger) HandleCommunityCancelRequestToJoin(state *ReceivedMessageSta
 	}
 
 	requestToJoin, err := m.communitiesManager.HandleCommunityCancelRequestToJoin(signer, cancelRequestToJoinProto)
+
+	if err != nil {
+		return err
+	}
+
+	community, err := m.communitiesManager.GetByID(cancelRequestToJoinProto.CommunityId)
+
 	if err != nil {
 		return err
 	}
 
+	senderID := state.CurrentMessageState.Contact.ID
+	senderPubKey, err := common.HexToPubkey(senderID)
+
+	if err != nil {
+		return err
+	}
+
+	if community.HasMember(senderPubKey) {
+		response, err := m.RemoveUserFromCommunity(
+			community.ID(),
+			senderID,
+		)
+		if err != nil {
+			return err
+		}
+
+		if len(response.Communities()) > 0 {
+			state.Response.AddCommunity(response.Communities()[0])
+		}
+	}
```

Comment:
I am wondering why not to put this logic directly into `communitiesManager.HandleCommunityCancelRequestToJoin`?

We would avoid double community reading from db -> `community, err := m.communitiesManager.GetByID(cancelRequestToJoinProto.CommunityId)`

Reply:
I refactored it to only fetch the community data once.